src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs

```diff
@@ -231,7 +231,7 @@ private bool IsUnusedParameterCandidate(IParameterSymbol parameter)
                 // TODO: Remove this when implicit operations are synthesised: https://github.com/dotnet/roslyn/issues/47829 
                 var methodSyntax = method.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax();
                 if (method.IsConstructor() &&
-                    _compilationAnalyzer.IsRecordDeclaration(methodSyntax))
+                    (_compilationAnalyzer.IsRecordDeclaration(methodSyntax) || _compilationAnalyzer.IsClassDeclaration(methodSyntax)))
```

Comment:
Where do we make sure this is a primary constructor and not a regular class constructor? Can you add tests to ensure that unused parameters are still flagged for a regular user written constructor in a regular class declaration?

Reply:
Ah, I see how this will work. Constructor symbol with its syntax node as a class declaration syntax guarantees it being a primary constructor. Would still be good to add a test for unused parameters of a regular constructor (if one doesn’t already exist)

Reply:
i don't love that we're testing this way.  I'd prefer it be done similar to the extension we have for GetAssociatedSynthesizedRecordProperty.  Can you make a similar extension for this question (i.e. 'IsPrimaryConstructor').